schemas/rum/action-schema.json

```diff
@@ -60,14 +60,24 @@
               },
               "readOnly": true
             },
-            "frustration_type": {
-              "type": "array",
-              "description": "Action frustration types",
-              "readOnly": true,
-              "items": {
-                "type": "string",
-                "enum": ["rage", "dead", "error"]
-              }
+            "frustration": {
+              "type": "object",
+              "description": "Action frustration properties",
+              "required": [
+                "type"
+              ],
+              "properties": {
+                "type": {
+                  "type": "array",
+                  "description": "Action frustration types",
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏idk if it was previously considered but since it is an array, should not it be `frustration.types`?

Reply:
It was not discussed, but IMO the singular version is fine:
* the fact that it is an array won't be user-facing too much, as most of the time the frustation will be rendered in a custom way by the UI
* when doing a query the user can specify on type at a time (ex: `@action.frustration.type:error`). Having `@action.frustration.types:error` looks a bit weird to me.

Reply:
ok, lgtm